Sources/ScipioKit/ProjectGenerator/BuildFlagsGenerator.swift

```diff
@@ -0,0 +1,134 @@
+import Foundation
+import PackageModel
+import PackageGraph
+import TSCBasic
+
+struct BuildFlagsGenerator {
+    private let rootPackage: Package
+    private let target: ResolvedTarget
+    private let buildConfiguration: BuildConfiguration
+    private let platforms: Set<SDK>
+
+    enum Error: LocalizedError {
+        case unsupportedSettings(TargetBuildSettingDescription.Setting)
+
+        var errorDescription: String? {
+            switch self {
+            case .unsupportedSettings(let setting):
+                return "Unsupported settings \(setting.kind) for \(setting.tool)"
+            }
+        }
+    }
+
+    init(package: Package, target: ResolvedTarget, buildConfiguration: BuildConfiguration, platforms: Set<SDK>) {
+        self.rootPackage = package
+        self.target = target
+        self.buildConfiguration = buildConfiguration
+        self.platforms = platforms
+    }
+
+    func generate() throws -> [String: XCConfigValue] {
+        guard let subPackage = rootPackage.graph.packages.first(where: { $0.targets.contains { $0.name == target.name } }),
+              let targetDescription = subPackage.manifest.targets.first(where: { $0.name == target.name }) else {
+            return [:]
```

Comment:
I think it should error by user of scipio, how about throwing error?

Reply:
Make sense. I fixed https://github.com/giginet/Scipio/pull/37/commits/439a85b4fdef19e791b3c35116ac528affe6a4f5

@freddi-kit